tutor/src/models/chapter-section.ts

```diff
@@ -7,11 +7,13 @@ export default class ChapterSection extends BaseModel {
 
     @readonly static blank = new ChapterSection()
 
-    constructor(arg: [number, number] | string = '') {
+    constructor(arg: ChapterSection | [number, number] | string = '') {
         super();
         modelize(this);
         if (isArray(arg)) {
             [this.chapter, this.section] = arg;
+        } else if (arg instanceof ChapterSection) {
+            [this.chapter, this.section] = arg.split();
         } else {
```

Comment:
yep, `instanceof` is pretty standard, TS will notice you used it and then treat everything in the block as that type.   There's also the concept of [type guards](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#typeof-type-guards)  if found ourselves doing this a lot we could make a `isChapterSection` or something like that